applications/FreeSurfaceApplication/custom_utilities/edgebased_levelset.h

```diff
@@ -40,1160 +36,1082 @@
+            mr_matrix_container.SetToZero(mA);
+            mB.resize(n_nodes);
+            mr_matrix_container.SetToZero(mB);
+            mStrVel.resize(n_nodes);
+            mr_matrix_container.SetToZero(mStrVel);
+
+            mdiv_error.resize(n_nodes);
+            mr_matrix_container.SetToZero(mdiv_error);
+            mdiag_stiffness.resize(n_nodes);
+            mr_matrix_container.SetToZero(mdiag_stiffness);
+            mis_slip.resize(n_nodes);
+            //	    ValuesVectorType external_pressure;
+            //	    external_pressure.resize(n_nodes);
+
+            // read velocity and pressure data from Kratos
+            mr_matrix_container.FillScalarFromDatabase(VISCOSITY, mViscosity, mr_model_part.Nodes());
+            mr_matrix_container.FillVectorFromDatabase(VELOCITY, mvel_n1, mr_model_part.Nodes());
+            mr_matrix_container.FillScalarFromDatabase(PRESSURE, mPn1, mr_model_part.Nodes());
+            mr_matrix_container.FillOldScalarFromDatabase(PRESSURE, mPn, mr_model_part.Nodes());
+            mr_matrix_container.FillOldVectorFromDatabase(VELOCITY, mvel_n, mr_model_part.Nodes());
+            mr_matrix_container.FillCoordinatesFromDatabase(mx, mr_model_part.Nodes());
+            // set flag for first time step
+            mFirstStep = true;
+
+            // loop to categorize boundary nodes
+            std::vector<unsigned int> tempFixedVelocities;
+            std::vector<array_1d<double, TDim>> tempFixedVelocitiesValues;
+            std::vector<unsigned int> tempPressureOutletList;
+            for (ModelPart::NodesContainerType::iterator inode = mr_model_part.NodesBegin();
+                 inode != mr_model_part.NodesEnd();
+                 inode++)
             {
-                if (inode->IsFixed(VELOCITY_Y) == false || inode->IsFixed(VELOCITY_Z) == false)
+                int index = inode->FastGetSolutionStepValue(AUX_INDEX);
+
+                if (inode->Is(INLET))
                 {
-                    std::cout << "error found on the fixity of node " << inode->Id() << std::endl;
-                    KRATOS_THROW_ERROR(std::logic_error, "velocities can be either all fixed or none fixed", "")
+                    tempFixedVelocities.push_back(index);
+                    tempFixedVelocitiesValues.push_back(mvel_n1[index]);
                 }
 
-
-                tempFixedVelocities.push_back(index);
-                tempFixedVelocitiesValues.push_back(mvel_n1[index]);
+                if (inode->Is(OUTLET))
+                {
+                    tempPressureOutletList.push_back(index);
+                    //		    mPressureOutlet.push_back(external_pressure[index]);
+                }
             }
+            mFixedVelocities.resize(tempFixedVelocities.size(), false);
+            mFixedVelocitiesValues.resize(tempFixedVelocitiesValues.size(), false);
+            mPressureOutletList.resize(tempPressureOutletList.size(), false);
 
-            if (inode->IsFixed(PRESSURE))
```

Comment:
One comment about this. Regardless I'm not aware of what this process does, in this case I'd keep the fixity check rather than using the flag outlet. My point is that in here you're assuming that all the nodes with a fixed `PRESSURE` DOF will be always be flagged as `OUTLET`, something that is not necessarily true.

Reply:
Think about the case we were discussing the other day, a pressure-based inlet.

Reply:
Done